modules/react/layout/lib/utils/grid.ts

```diff
@@ -79,14 +105,22 @@ const gridProps = {
  * );
  *
  */
-export function grid<P extends GridStyleProps>(props: P) {
-  const styles = {};
+export function grid<P extends SpaceStyleProps>(props: P) {
+  let styles = {};
   for (const key in props) {
-    if (key in gridProps) {
-      const attr = gridProps[key as keyof GridStyleProps];
-      const value = props[key];
-      // @ts-ignore TS doesn't like adding a potentially unknown key to an object, but because we own this object, it's fine.
-      styles[attr] = value;
+    if (props.hasOwnProperty(key)) {
+      if (key in gridProps) {
+        const attr = gridProps[key as keyof GridStyleProps];
+        const value = props[key];
+        // @ts-ignore TS doesn't like adding a potentially unknown key to an object, but because we own this object, it's fine.
+        styles[attr] = value;
+      }
+      if (key in GridSpaceStyleProps) {
+        const value = props[key as keyof GridSpaceProps] as GridSpacePropsValues;
+        const spaceFn = GridSpaceStyleProps[key as keyof GridSpaceProps];
+        const style = spaceFn(value);
+        styles = {...styles, ...style};
+      }
```

Comment:
Looks good. Let's add tests and docs for this next. 👍🏻

Reply:
Added some tokens in the existing tests and added to docs.